sdks/python/pytest.ini

```diff
@@ -70,6 +70,7 @@ markers =
     uses_mock_api: tests that uses the mock API cluster.
     uses_feast: tests that uses feast in some way
     gemini_postcommit: gemini postcommits that need additional deps.
+    require_non_self_hosted: tests which can't be run on Beam's self hosted GitHub Actions runners. Usually this is because of issues with docker-in-docker or other environmental challenges. Context: https://github.com/apache/beam/pull/35585
```

Comment:
we try to decouple GitHub Actions with the other part of Beam code, so developers can run tests on their machines. Can we rename  "require_non_self_hosted" to something like require_docker_in_docker or other informative marker names that does not assume the tests run in GitHub Action ?